test/functional/admin/worldwide_offices_controller_test.rb

```diff
@@ -306,6 +304,58 @@ class Admin::WorldwideOfficesControllerTest < ActionController::TestCase
     assert_not ContactNumber.exists?(contact_number.id)
   end
 
+  test "PUT :update deletes contact numbers that have a blank label" do
+    worldwide_organisation, office = create_worldwide_organisation_and_office
+    contact_number = office.contact.contact_numbers.create!(label: "Phone", number: "1234")
+
+    put :update,
+        params: {
+          worldwide_office: {
+            contact_attributes: {
+              id: office.contact.id,
+              title: "Head office",
+              contact_numbers_attributes: {
+                "0" => {
+                  id: contact_number.id,
+                  label: "",
+                  number: contact_number.number,
+                },
+              },
+            },
+          },
+          id: office,
+          worldwide_organisation_id: worldwide_organisation,
+        }
+
+    assert_not ContactNumber.exists?(contact_number.id)
+  end
+
+  test "PUT :update deletes contact numbers that have a blank number" do
+    worldwide_organisation, office = create_worldwide_organisation_and_office
+    contact_number = office.contact.contact_numbers.create!(label: "Phone", number: "1234")
+
+    put :update,
+        params: {
+          worldwide_office: {
+            contact_attributes: {
+              id: office.contact.id,
+              title: "Head office",
+              contact_numbers_attributes: {
+                "0" => {
+                  id: contact_number.id,
+                  label: contact_number.label,
+                  number: "",
+                },
+              },
+            },
+          },
+          id: office,
+          worldwide_organisation_id: worldwide_organisation,
+        }
+
+    assert_not ContactNumber.exists?(contact_number.id)
```

Comment:
We shouldn't be deleting the record if a user leaves one field blank (as they will lose the data they've entered into the other field).  Instead there should be a validation error displayed (so change the `&&` to `||` in the controller, or switch back to using `if` but with `blank?` instead of `empty?`, as `nil` responds to that method).

Reply:
This should all be solved now